### PR TITLE
feat(rule): get evaluateEvent from events with valid move-type [CARROT-1032]

### PR DESCRIPTION
**Pull request by AMarcosCastelo**
### Summary

Get rule subject from events with move-type metadata defined.

### Related links

https://app.clickup.com/t/3005225/CARROT-1032

---

- [x] **I, the PR author, declare that this PR works as expected and does not break any service. I also declare that I gave my best to apply all of the best practices and that I'm leaving the code better than I found it.**

## Summary by CodeRabbit

- **New Features**
  - Expanded event types to include 'Shipment-request' in addition to 'Pick-up' for address verification.

- **Refactor**
  - Enhanced `SameSourceAndPickUpAddressesProcessor` for better functionality and readability.
  - Updated initialization of constants and event creation logic to support new event types.

### CODE REVIEW

File: apps/methodologies/bold/rule-processors/mass/same-source-and-pick-up-addresses/src/lib/same-source-and-pick-up-addresses.processor.ts [hunk 1]
```diff
@@ -1,90 +1,80 @@
...
-  static resultComments = {
-    doNotMatch: 'Source and pick-up addresses do not match',
-    noSourceActorEvent: 'No actor event with source actor type found',
-    ruleNotApplicable:
+export class SameSourceAndPickUpAddressesProcessor extends ParentDocumentRuleProcessor<RuleSubject> {
+  private ResultComments = {
+    DO_NOT_MATCH: 'Source and pick-up addresses do not match',
+    NO_SOURCE_ACTOR_EVENT: 'No actor event with source actor type found',
+    RULE_NOT_APPLICABLE:
       'Rule not applicable: No move event with pick-up move type found',
```
Comment (RafaPalau) [inline apps/methodologies/bold/rule-processors/mass/same-source-and-pick-up-addresses/src/lib/same-source-and-pick-up-addresses.processor.ts]: Acho que teria que ser algo como:
```suggestion
      'Rule not applicable: No move event with pick-up or Shipment-request move type found',
```

**Review by RafaPalau — DISMISSED** [review]

File: apps/methodologies/bold/rule-processors/mass/same-source-and-pick-up-addresses/src/lib/same-source-and-pick-up-addresses.processor.ts (hunk 1 above)
Comment (AMarcosCastelo) [reply apps/methodologies/bold/rule-processors/mass/same-source-and-pick-up-addresses/src/lib/same-source-and-pick-up-addresses.processor.ts]: [feature/update-same-source-and-pick-up-address 1f4cd19] chore: update the same-source-and-pick-up-address rule

File: apps/methodologies/bold/rule-processors/mass/same-source-and-pick-up-addresses/src/lib/same-source-and-pick-up-addresses.processor.e2e.spec.ts [hunk 2]
```diff
@@ -71,7 +70,7 @@ testRuleProcessorWithMassDocuments(
         ]);
       });
 
-      it('should return returnValue true when neither OPEN nor move event is present', async () => {
+      it('should REJECTE when the address ids do not match', async () => {
```
Comment (cris-santos) [inline apps/methodologies/bold/rule-processors/mass/same-source-and-pick-up-addresses/src/lib/same-source-and-pick-up-addresses.processor.e2e.spec.ts:73]:
```suggestion
      it('should return REJECTED when the address ids do not match', async () => {
```

File: apps/methodologies/bold/rule-processors/mass/same-source-and-pick-up-addresses/src/lib/same-source-and-pick-up-addresses.processor.ts [hunk 3]
```diff
@@ -1,90 +1,80 @@
...
+      evaluateEvent.address.id === sourceActorEvent.address.id
+        ? RuleOutputStatus.APPROVED
+        : RuleOutputStatus.REJECTED;
+
+    return {
+      resultStatus,
+      ...(resultStatus === RuleOutputStatus.REJECTED && {
+        resultComment: this.ResultComments.DO_NOT_MATCH,
+      }),
+    };
```
Comment (cris-santos) [inline apps/methodologies/bold/rule-processors/mass/same-source-and-pick-up-addresses/src/lib/same-source-and-pick-up-addresses.processor.ts:58]: ao invés de dois ternários iguais, uma alternativa seria ter um if só `if (evaluateEvent.address.id === sourceActorEvent.address.id)`

File: apps/methodologies/bold/rule-processors/mass/same-source-and-pick-up-addresses/src/lib/same-source-and-pick-up-addresses.processor.ts [hunk 4]
```diff
@@ -1,90 +1,80 @@
...
-    noSourceActorEvent: 'No actor event with source actor type found',
-    ruleNotApplicable:
-      'Rule not applicable: No move event with pick-up move type found',
+export class SameSourceAndPickUpAddressesProcessor extends ParentDocumentRuleProcessor<RuleSubject> {
+  private ResultComments = {
+    DO_NOT_MATCH: 'Source and pick-up addresses do not match',
+    NO_SOURCE_ACTOR_EVENT: 'No actor event with source actor type found',
+    RULE_NOT_APPLICABLE:
+      'Rule not applicable: No event with pick-up or Shipment-request move-type found',
   };
```
Comment (cris-santos) [inline apps/methodologies/bold/rule-processors/mass/same-source-and-pick-up-addresses/src/lib/same-source-and-pick-up-addresses.processor.ts:28]: pra padronizar, uma sugestão seria sempre usar o value dos atributos, que aí tbm facilita na hora de identificar, ex: "Pick-up", "actor-type", etc

**Review by cris-santos — APPROVED** [review]

**Review by RafaPalau — APPROVED** [review]

**Pull request merged**